### PR TITLE
feat(aws-s3-storage): set ACL of tarball files

### PR DESCRIPTION
**Pull request by favoyang**
**Type:** feature 
**Scope:** plugins/aws-s3-storage

The following has been addressed in the PR:

*  There is a related issue? #250 

*  Unit or Functional tests are included in the PR? No.

**Description:**

The supportive PR introduces *S3Config.tarballACL* to specify ACL (access-control list) of tarball files.
- set to 'private' by default, same as the current behavior.
- when set to 'public-read', it will grant tarball files anonymous read permission, which is necessary to enable S3 integrated CDN service like Amazon CloudFront.

### CODE REVIEW

**Comment by favoyang** [comment]
Based on the reconsideration of https://github.com/verdaccio/verdaccio/pull/1580#issuecomment-578422556, this PR shall be updated to only keep the logic of ACL, and remove the logic of tarballEdgeUrl.

File: plugins/aws-s3-storage/src/s3PackageManager.ts [hunk 1]
```diff
@@ -242,15 +245,20 @@ export default class S3PackageManager implements ILocalPackageManager {
...
+              's3: [S3PackageManager writeTarball headObject] @{convertedErr} non a 404 emit error'
+            );
 
             uploadStream.emit('error', convertedErr);
           } else {
             this.logger.debug('s3: [S3PackageManager writeTarball managedUpload] init stream');
-            const managedUpload = this.s3.upload(Object.assign({}, baseS3Params, { Body: uploadStream }));
+            const managedUpload = this.s3.upload(
+              Object.assign({}, baseS3Params, { Body: uploadStream, ACL: this.tarballACL })
+            );
```
Comment (favoyang) [inline plugins/aws-s3-storage/src/s3PackageManager.ts]: This is the core logic of this PR, passing tarballACL when uploading a tarball file.

**Comment by favoyang** [comment]
@juanpicado the PR #249 has been simplified to only include the feature of setting ACL of tarball files. Though it is a supportive PR for #250, but it offering an independent feature, I think it can be reviewed and merged individually. 

Please reconsider to remove the `do-not-merge` label. To review this, I suggest you read the updated description of this PR, #250.

**Comment by juanpicado** [comment]
@favoyang sorry the dealy. I'm ready to merge this. Could you rebase form master and run `pnpm changestet`?
https://github.com/verdaccio/monorepo/blob/9.x/CONTRIBUTING.md#adding-a-changeset

**Review by juanpicado — APPROVED** [review]
LGTM 👍🏼

**Comment by favoyang** [comment]
@juanpicado, so I should migrate the PR to verdaccio/verdaccio@master then re-submit a new PR?

**Comment by juanpicado** [comment]
No for now, just thinking to merge on 5.x under a flag and let see how it goes. I want to change the plugins architecture (help ls welcome) in master is to early to put this feature there.

**Comment by favoyang** [comment]
Sorry that I’m still confused. The 5.x branch on verdaccio seems has no packages folder, and the master branch on monorepo is basically empty. So what should I do next to “rebase from master and run pnpm”?

**Comment by juanpicado** [comment]
> Sorry that I’m still confused. The 5.x branch on verdaccio seems has no packages folder, and the master branch on monorepo is basically empty. So what should I do next to “rebase from master and run pnpm”?

No problem, sorry I was in a hurry when I wrote.

This PR would be great if you 
- `git rebase origin/9.x` (it should not have any conflict). 
- Run `pnpm install`
- Run `pnpm changeset` and follow the CONTRIBUTION guidelines to create the changeset.

Once it is done, I will merge, the GitHub Action will detect your changeset and create a PR which I'll merge and then the new version will be published in npmjs.

If you like you can also copy this change to https://github.com/verdaccio/verdaccio/tree/master/packages/plugins/aws-storage but this is optional.

**Comment by favoyang** [comment]
> git rebase origin/9.x (it should not have any conflict).

Not sure for what reasons run `rebase 9.x` results in conflicts. So I run `merge 9.x` instead. You should be able to merge this PR with the squash option to get a clean changeset.

> Run pnpm install

Done

```
$ pnpm --version
6.0.1
$ pnpm install
Scope: all 14 workspace projects
Already up-to-date
WARN  Failed to find "/fsevents/1.2.13" in lockfile during hoisting. Next aliases will not be hoisted: fsevents
```

> Run pnpm changeset and follow the CONTRIBUTION guidelines to create the changeset. (This will create a markdown file on .changeset folder which you can add enough information which will be part of the CHANGELOG`.

```
 $ pnpm changeset

> @verdaccio/monorepo@0.0.0 changeset /home/favo/projects/verdaccio-test/monorepo
> changeset

🦋  Which packages would you like to include? … verdaccio-aws-s3-storage
  ✔ verdaccio-aws-s3-storage
```

Type/select the `verdaccio-aws-s3-storage` package, then press enter results in error:

```
🦋  Which packages would you like to include? · No items were selected
🦋  error You must select at least one package to release
🦋  error (You most likely hit enter instead of space!)
```

**Review by juanpicado — APPROVED** [review]

**Pull request merged**